guides/rest_api.md

```diff
@@ -0,0 +1,44 @@
+# REST API
+
+## Authentication
+
+To make full use of the API you need to authenticate as an existing user. A successful authentication returns a *token* that you can use in subsequent requests to authenticate yourself as that user. You do this by setting the token as request header.
+
+> curl -H "Authorization: Bearer AUTH-TOKEN" http://localhost:4000/api/rest/v1/
+
+```
+POST /api/rest/v1/auth
```

Comment:
I don't know how rest apis usually handle this, but i would go for `/api/rest/v1/auth/token` instead and keep the top level clear.

Reply:
In general, the URL represents the resource and the HTTP method is the _verb_ acting on the resource.

```
GET    /networks lists networks
GET    /networks/1 shows network with id 1
POST   /networks creates a new network
PUT    /networks/1 updates the network
DELETE /networks/1 deletes the network
```

Authentication is a weird one in REST terms as you don't really create something so it's hard to fit into this schema. Maybe a session. We had a discussion about this a while ago https://github.com/podlove/radiator/pull/29#discussion_r272470397

The most REST-ish way I see is POST to `/sessions` to create a session to log in. Not sure about prolonging, maybe a PUT to `/sessions` even though that's not exactly following the REST schema.

Reply:
Just saw that Phoenix has explicit support for what they call [Singleton Resources](https://hexdocs.pm/phoenix/1.4.8/Phoenix.Router.html#resources/4-singleton-resources). In their example for the user profile. But could be used for sessions as well.